src/main/java/org/openstreetmap/atlas/geography/atlas/builder/proto/ProtoAtlasBuilder.java

```diff
@@ -0,0 +1,237 @@
+        writeLinesToBuilder(atlas, protoAtlasBuilder);
+
+        final ProtoAtlasContainer protoAtlas = protoAtlasBuilder.build();
+        resource.writeAndClose(protoAtlas.toByteArray());
+    }
+
+    private List<ProtoTag> buildProtoTagListFromTagMap(final Map<String, String> tags)
+    {
+        final List<ProtoTag> protoTags = new ArrayList<>();
+        for (final Map.Entry<String, String> entry : tags.entrySet())
+        {
+            final ProtoTag.Builder tagBuilder = ProtoTag.newBuilder();
+
+            // TODO what happens if entry.getValue() returns null?
+            // this could happen if someone inserted a null entry at that key
+            // maybe use entry.getValue() != null ? entry.getValue() : ""
+            final String fullTag = entry.getKey() + TAG_DELIMITER + entry.getValue();
+            tagBuilder.setTagText(fullTag);
+            protoTags.add(tagBuilder.build());
+        }
+        return protoTags;
+    }
+
+    private ProtoLocation convertLocationToProtoLocation(final Location location)
+    {
+        final ProtoLocation.Builder protoLocationBuilder = ProtoLocation.newBuilder();
+        protoLocationBuilder.setLatitude(Math.toIntExact(location.getLatitude().asDm7()));
+        protoLocationBuilder.setLongitude(Math.toIntExact(location.getLongitude().asDm7()));
+        return protoLocationBuilder.build();
+    }
+
+    private Location convertProtoLocationToLocation(final ProtoLocation protoLocation)
+    {
+        final Longitude longitude = Longitude.dm7(protoLocation.getLongitude());
+        final Latitude latitude = Latitude.dm7(protoLocation.getLatitude());
+        return new Location(latitude, longitude);
+    }
+
+    private void parseLines(final PackedAtlasBuilder builder, final List<ProtoLine> lines)
+    {
+        lines.forEach(line ->
+        {
+            final long identifier = line.getId();
+            final List<Location> shapePoints = line.getShapePointsList().stream()
+                    .map(this::convertProtoLocationToLocation).collect(Collectors.toList());
+            final PolyLine geometry = new PolyLine(shapePoints);
+            final Map<String, String> tags = parseTags(line.getTagsList());
+            builder.addLine(identifier, geometry, tags);
+        });
+    }
+
+    private void parsePoints(final PackedAtlasBuilder builder, final List<ProtoPoint> points)
+    {
+        points.forEach(point ->
+        {
+            final long identifier = point.getId();
+            final Longitude longitude = Longitude.dm7(point.getLocation().getLongitude());
+            final Latitude latitude = Latitude.dm7(point.getLocation().getLatitude());
+            final Location geometry = new Location(latitude, longitude);
+            final Map<String, String> tags = parseTags(point.getTagsList());
+            builder.addPoint(identifier, geometry, tags);
+        });
+    }
+
+    private Map<String, String> parseTags(final List<ProtoTag> tagsList)
```

Comment:
Perhaps this class can go in the Converter package?

Reply:
Same here.